mito-ai/src/utils/notebook.tsx

```diff
@@ -339,4 +340,122 @@ export const scrollToCell = (
 
     // Use the new JupyterLab scrollToCell method instead of DOM node scrollIntoView
     void notebookPanel.content.scrollToCell(cell, position);
+}
+
+export const scrollToNextCellWithDiff = (
+    notebookTracker: INotebookTracker,
+    currentCellId: string,
+    changedCells: { cellId: string, originalCode: string, currentCode: string }[],
+): void => {
+    // Early return if no diffs remain
+    if (changedCells.length === 0) {
+        return;
+    }
+
+    // Convert changedCells to agentEdits format for internal use
+    const agentEdits = changedCells.map(change => ({
+        cellId: change.cellId,
+        code: change.currentCode
+    }));
+
+    // Find current cell's position in the edits list
+    const currentEditIndex = agentEdits.findIndex(edit => edit.cellId === currentCellId);
+    if (currentEditIndex < 0) {
+        return; // Current cell not found in edits
+    }
+
+    // Find the next cell that still has a diff
+    const nextCellWithDiff = agentEdits
+        .slice(currentEditIndex + 1)
+        .find(edit => changedCells.some(change => change.cellId === edit.cellId));
+
+    if (!nextCellWithDiff) {
+        return; // No more cells with diffs
+    }
+
+    // Scroll to and select the next cell with diff
+    setActiveCellByID(notebookTracker, nextCellWithDiff.cellId);
```

Comment:
I think we should be using the notebookPanel here. Reminder, pretty much whenever we are using the Agent mode, let's use the notebookPanel which we store at the beginning of agent mode in the ref `agentTargetNotebookPanelRef`